siege_utilities/geo/gazetteers/nominatim_gazetteer.py

```
@@ -0,0 +1,373 @@
+        self._user_agent = user_agent
+        self._server_url = server_url
+        self._timeout = timeout
+        self._rate_limit = _SELFHOSTED_RATE_LIMIT_S if server_url else _PUBLIC_RATE_LIMIT_S
+        self._client = self._build_client()
+        # Per-instance cache keyed on the lookup tuple so different hint
+        # combinations don't collide.
+        self._cached_lookup = functools.lru_cache(maxsize=cache_size)(
+            self._uncached_lookup
+        )
+
+    def _build_client(self) -> Any:
```

Ruff (0.15.12)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `_build_client`

(ANN401)